src/main/java/uk/gov/hmcts/ccd/domain/service/createevent/AuthorisedCreateEventOperation.java

```diff
@@ -87,6 +96,74 @@ public CaseDetails createCaseEvent(String caseReference,
+            createCaseEventDetails.getCaseTypeDefinition(),
+            createCaseEventDetails.getAccessProfiles());
+
+        checkCaseCategoryId(createCaseEventDetails.getCaseTypeDefinition(), categoryId);
+
+        if (!version.equals(createCaseEventDetails.getCaseDetails().getVersion())) {
+            throw new BadRequestException("003 Wrong CaseVersion");
+        }
+
+        final CaseDetails caseDetails = createEventOperation.createCaseSystemEvent(caseReference,
+            version, attributePath, categoryId);
+        return verifyReadAccess(createCaseEventDetails.getCaseTypeDefinition(), createCaseEventDetails
+            .getAccessProfiles(), caseDetails);
+    }
+
+    private void checkCaseCategoryId(CaseTypeDefinition caseTypeDefinition, String categoryId) {
+        Boolean validCategoryId = categoryId == null || caseTypeDefinition.getCategories()
+            .stream()
+            .map(cfd -> cfd.getCategoryId())
+            .filter(value -> StringUtils.isNotBlank(value))
+            .anyMatch(value ->  value.equals(categoryId));
+        if (!validCategoryId) {
+            throw new BadRequestException("002 Invalid categoryId");
+        }
+    }
+
+    private void checkCaseDocumentData(String attributePath, CreateCaseEventDetails createCaseEventDetails) {
+        try {
+            if (!caseDetailsJsonParser.containsDocumentUrl(createCaseEventDetails.getCaseDetails(), attributePath)) {
+                throw new BadRequestException("Field denoted by path: '" + attributePath
+                    + "' is not a document field type");
+            }
+        } catch (PathNotFoundException e) {
+            throw  new BadRequestException("Field '" + attributePath + "' cannot be found");
+        }
+    }
+
+    private CreateCaseEventDetails getCreateCaseEventDetails(String caseReference) {
+        CreateCaseEventDetails createCaseEventDetails = new CreateCaseEventDetails();
+        CaseDetails existingCaseDetails = getCaseOperation.execute(caseReference)
+            .orElseThrow(() -> new ResourceNotFoundException("Case not found"));
+        createCaseEventDetails.setCaseDetails(existingCaseDetails);
+
+        Set<AccessProfile> accessProfiles = caseAccessService.getAccessProfilesByCaseReference(caseReference);
+        if (accessProfiles == null || accessProfiles.isEmpty()) {
+            throw new ValidationException("Cannot find user roles for the user");
+        }
+        createCaseEventDetails.setAccessProfiles(accessProfiles);
+
+        String caseTypeId = existingCaseDetails.getCaseTypeId();
+        final CaseTypeDefinition caseTypeDefinition = caseDefinitionRepository.getCaseType(caseTypeId);
+        if (caseTypeDefinition == null) {
+            throw new ValidationException("Cannot find case type definition for  " + caseTypeId);
+        }
+        createCaseEventDetails.setCaseTypeDefinition(caseTypeDefinition);
+        return createCaseEventDetails;
+    }
```

Comment:
This is the same as the start of `createCaseEvent()` in this class - we could replace L#77-89 by calling this?

I think we should also include the (new) `verifyCaseAccessCategories()` that has been added since this PR was started, on L#91, as part of this private method.